### ISSUE TITLE
converging.md

### ISSUE THREAD

**Issue by githubsgi** [pytorch/torchtitan#831]
In the [page](https://github.com/pytorch/torchtitan/blob/main/docs/converging.md) . Can someone please clarify the the following.

1.  How many (dp) and what type of GPU was used for the [chart](https://github.com/pytorch/torchtitan/blob/main/docs/converging.md#test-results). 
2. What is FSDP 8 , 8 GPU's or FP 8 ?
3.

**Comment by mori360** [comment]
> How many (dp) and what type of GPU was used for the [chart](https://github.com/pytorch/torchtitan/blob/main/docs/converging.md#test-results).

They all have `dp=8`, we run with H100, you can find in the [doc ](https://github.com/pytorch/torchtitan/blob/3996b634a58c7b7de3f7f7b815058245d8c69e01/docs/performance.md?plain=1#L3-L7)here.

> What is FSDP 8 , 8 GPU's or FP 8 ?

Yeah

**Comment by tianyu-l** [comment]
> How many (dp)

The set of jobs use 8 to 64 GPUs, the breakdown specified in the table.
e.g. FSDP 8, TP 2, CP 2, PP 2
means FSDP degree 8 * Tensor Parallel degree 2 * Context Parallel degree 2 * Pipeline Parallel degree 2

> What is FSDP 8 , 8 GPU's or FP 8 ?

FSDP 8 means Fully Sharded Data Parallel degree is 8
Here is a doc for FSDP https://github.com/pytorch/torchtitan/blob/main/docs/fsdp.md

**Comment by githubsgi** [comment]
Thanks,  so FSDP 8  means  dividing  params, gradients and optimizer states into 8 parts ?    

Is FSDP 8, TP 2, CP 2, PP 2   different from TP 2 , FSDP 8, , PP 2,  CP 2 ?

**Comment by mori360** [comment]
> Thanks, so FSDP 8 means dividing params, gradients and optimizer states into 8 parts ?

Yeah

> Is FSDP 8, TP 2, CP 2, PP 2 different from TP 2 , FSDP 8, , PP 2, CP 2 ?

There' no difference for the order when we call them (as well as the config order in CLI)
TorchTitan processes these parallels in a specific order, that won't be affected by the user inputs.

**Comment by githubsgi** [comment]
Thanks again.   In case of  FSDP 8, TP 2, CP 2, PP 2  , what is the specific order  ?    Is there a way to trace the collectives of the 4 dimensions ?

**Comment by tianyu-l** [comment]
@githubsgi 
> Thanks again. In case of FSDP 8, TP 2, CP 2, PP 2 , what is the specific order ?

There are two concepts of "order":
- the order of device mesh from outer to inner is specified here: https://github.com/pytorch/torchtitan/blob/main/torchtitan/parallelisms/parallel_dims.py#L53. In our case, we used PP -> DP -> CP -> TP
- the order of applying parallelisms can be found in train.py

> Is there a way to trace the collectives of the 4 dimensions ?

You can look at the profiler trace (with tools like perfetto), after dumping them via https://github.com/pytorch/torchtitan/blob/main/torchtitan/config_manager.py#L92

**Comment by githubsgi** [comment]
Thanks  - ["pp", "dp_replicate", "dp_shard", "cp", "tp"], is the order.   Is dp_replicate  just DDP ?   I guess pp, dp_shard, cp and tp can co-exist.  Not sure what dp_replicate can co-exist with.

**Comment by githubsgi** [comment]
Follow on questions on the outer to inner of PP -> DP -> CP -> TP .   Supposing I have 3,4,1,8  for PP,DP,CP, and PP respectively,  Would the following be correct then.  Ranks 0 -7 is in one TP group. Ranks 0,8,16,24  are  in one DP group.  Ranks 0, 32,and 64 are in one PP group.    Is it possible to change the order to  say DP->CP->PP->TP ?

**Comment by tianyu-l** [comment]
@githubsgi 
Your interpretation looks right to me.

> Is it possible to change the order to say DP->CP->PP->TP ?

I think so, you can try modifying the order here
https://github.com/pytorch/torchtitan/blob/main/torchtitan/distributed/parallel_dims.py#L56-L59
